### DISCUSSION TITLE
Installation issue - Cannot find type 'TaskPriority' in scope

### DISCUSSION THREAD

**Discussion by jmgawecki** [pointfreeco/swift-composable-architecture#831] [Q&A] [answered]
Hi, I am having an issue adding the TCA to me project/workspace. Xcode cannot find some things such as `TaskPriority`. But I can find it with the quick opening.

Some issues:
Cannot find type 'TaskPriority' in scope
'Task' cannot be constructed because it has no accessible initializers
Value of type 'Task' has no member 'cancel'
Cannot find type 'CancellationError' in scope
Subscript 'subscript(dynamicMember:)' requires that 'State' conform to 'Equatable'
Value of type 'StorePublisher<State>' has no dynamic member 'values' using key path from root type 'State'

All the Structs that cannot be find, are findable, and are public.

I am not sure if I understand what the problem is, long ago I've installed the same way to another project, with the SPM. Now there are some issues. Any suggestions? Kindly thank you.

**Comment by stephencelis** [comment]
Can you include some more information to help debug?

- What version of swift-composable-architecture is in your project?
- What version of Xcode are you using?
- What platform are you compiling for?
- What version of macOS are you running?

**Reply by jmgawecki** [reply]
Hi Stephen thanks for answering, sorry right away:
- TCA 0.9.0
- Xcode 13.0 Beta
- iOS 14.1
- MacOS 12.0 Beta

**Comment by siimon** [comment, marked as answer]
I had the same problem but I was running an earlier Xcode beta, upgrading to 13.0 (13A233) solved it for me

**Reply by jmgawecki** [reply]
What is the version you are currently running? I think I’m running the latest one as the terminal found no updates for Xcode

**Comment by JacksonUtsch** [comment]
Having similar issues on macOS beta. Started happening after installing Xcode 13.0 (13A233). Had to change toolchains (to 5.5 Release) to get Tasks to register for the compiler but now having other issues compiling TCA. Tried running on Xcode 13 beta but no luck so far.

Here's some of the issues I have: 

```.../checkouts/swift-composable-architecture/Sources/ComposableArchitecture/Beta/Concurrency.swift:222:10: Value of type 'StorePublisher<State>' has no dynamic member 'values' using key path from root type 'State'```

```.../checkouts/swift-composable-architecture/Sources/ComposableArchitecture/SwiftUI/ConfirmationDialog.swift:158:30: No type named 'Visibility' in module 'SwiftUI'```

Do Xcode toolchains differ from the Swift ones? Maybe having different Xcode installs messed up what the Xcode 13.0 toolchain actually is and Swift 5.5 release is missing some SwiftUI components?

Also..

• Latest TCA release (as of writing) 0.28.1
• Xcode 13 release
• macOS Monterey
• macOS Monterey (latest update)

**Reply by JacksonUtsch** [reply]
[This](https://github.com/swift-server/swift-service-lifecycle/issues/110) issue seems to contain the underlying issue for me. Oddly enough the 13.0 RC toolchain builds TCA but Swift 5.5 Release does not. Additionally using ```#if compiler(>=5.5) && canImport(_Concurrency)``` only passes the conditional on 5.5 and not the RC. So, this results in not being able to use the concurrency features with the TCA dependency until 5.5 is compatible.

These conditionals are 🙄

**Reply by stephencelis** [reply]
Were you able to track down the right way to get things building? Just want to make sure we don't need to make changes to TCA. We're not on Monterey yet so haven't been able to test that 😅

**Reply by jmgawecki** [reply]
I solved my issue by downloading Xcode 13 that is not a beta release, right from the appstore. I will leave the discussion open as I can see you have some undergoing discussion here. Good luck :) and thanks for TCA!

**Reply by JacksonUtsch** [reply]
I was not able to, I'll try some more things today. Glad @jmgawecki got it to work. Are you using any concurrency features though?

**Reply by JacksonUtsch** [reply]
Okay, so, Xcode RC is unusable because it doesn't include the macOS Monterey SDK. So have to revert to Xcode Beta and using 5.5 toolchain. Additionally TCA does not build with this unless you revert the changes from [this](https://github.com/pointfreeco/swift-composable-architecture/commit/63e5c430197a982b7127195fa7146ca6bbe538e3#diff-0be678bbb2d18e432c390d4d9627e8e823f9568cd3b7bdc2a6e0fb300ff54c62) commit calling `.values` from the `StorePublisher`. After those changes things compile and work.  😌
